provider.go

```diff
@@ -21,12 +21,12 @@ type Provider struct {
 
 // DefaultProviderRegistryHost is the hostname used for provider addresses that do
 // not have an explicit hostname.
-const DefaultProviderRegistryHost = svchost.Hostname("registry.terraform.io")
+const DefaultProviderRegistryHost = svchost.Hostname("registry.opentf.org")
 
 // BuiltInProviderHost is the pseudo-hostname used for the "built-in" provider
 // namespace. Built-in provider addresses must also have their namespace set
 // to BuiltInProviderNamespace in order to be considered as built-in.
-const BuiltInProviderHost = svchost.Hostname("terraform.io")
+const BuiltInProviderHost = svchost.Hostname("opentf.org")
```

Comment:
Are you 100% sure you want to be changing this?

I believe this would require many followup changes in the opentf codebase, though I might be wrong.

Reply:
Good question, are we keeping the builtins as referring to terraform.io?

if so, I can revert this bit.

Reply:
@Yantrio let's revert for now maybe, we can change it later once we better understand the ramifications.

Reply:
Looks like `BuiltInProviderHost` is used to determine whether a provider is builtin, and changing this behaviour would break existing logic

Reply:
Reverted. Tested this with the `terraform_data` resource and it works